### ISSUE TITLE
 [SC2252] False positive when using different variables

### ISSUE THREAD

**Issue by frdrk345** [koalaman/shellcheck#1891]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2252
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2252)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
if [ 0 -ne "${FOO}" ] || [ 0 -ne "${BAR}" ]; then
    echo hello
fi
```

#### Here's what shellcheck currently says:

In docker/external/minimal.sh line 2:
if [ 0 -ne "${FOO}" ] || [ 0 -ne "${BAR}" ]; then
                      ^-- SC2252: You probably wanted && here, otherwise it's always true.

For more information:
  https://www.shellcheck.net/wiki/SC2252 -- You probably wanted && here, othe...

#### Here's what I wanted or expected to see:

Not an error.
The wiki page mentions the special case when when the variables foo and bar is the same.
But in this case, I cannot see that this could apply.

Using FOO=0, BAR=0, respectivly FOO=1, BAR=1 must give different outcome for this if statement.

The example below works. So maybe there is some strange thing when I am using yoda-statements (backwards, always write: <constant> <compare-operator> <variable>).
Is this a bug?, I think so, or am I missing something.

Great tool BTW!

```sh
#!/bin/bash
if [ "${FOO}" -ne 0 ] || [ "${BAR}" -ne 0 ]; then
    echo hello
fi
```

**Comment by koalaman** [comment]
This was indeed just a pattern matching failure that relied on the more canonical order (Yoda comparisons don't make as much sense in sh where `[ x = y ]` doesn't assign). It should be fixed now. Thanks!
